src/main/java/com/google/devtools/build/lib/analysis/starlark/FunctionTransitionUtil.java

```diff
@@ -407,36 +409,42 @@ private static void updateOutputDirectoryNameFragment(
     CoreOptions buildConfigOptions = toOptions.get(CoreOptions.class);
     Set<String> updatedAffectedByStarlarkTransition =
         new TreeSet<>(buildConfigOptions.affectedByStarlarkTransition);
-    // Add newly changed native options to overall list of changed native options
+    // Add newly changed options to overall list of changed options
     for (String option : changedOptions) {
-      if (option.startsWith(COMMAND_LINE_OPTION_PREFIX)) {
-        updatedAffectedByStarlarkTransition.add(
-            option.substring(COMMAND_LINE_OPTION_PREFIX.length()));
-      }
+      updatedAffectedByStarlarkTransition.add(option);
     }
     buildConfigOptions.affectedByStarlarkTransition =
         ImmutableList.sortedCopyOf(updatedAffectedByStarlarkTransition);
-
-    // hash all relevant native option values;
     TreeMap<String, Object> toHash = new TreeMap<>();
-    for (String nativeOption : updatedAffectedByStarlarkTransition) {
-      Object value;
-      try {
-        value =
-            optionInfoMap
-                .get(nativeOption)
-                .getDefinition()
-                .getField()
-                .get(toOptions.get(optionInfoMap.get(nativeOption).getOptionClass()));
-      } catch (IllegalAccessException e) {
-        throw new RuntimeException(
-            "IllegalAccess for option " + nativeOption + ": " + e.getMessage());
+
+    // Hash all affected native options.
+    for (String optionName : buildConfigOptions.affectedByStarlarkTransition) {
```

Comment:
I wonder if `updatedAffectedByStarlarkTransition` would be more clear here. It's a bit subtle that `buildConfigOptions` gets mutated in this method, so depending on when you use it it may or may not have updated values. 

Just commenting for clarity - I believe the semantics are correct here.

Reply:
See my previous comment about extracting a new updateAffectedByStarlarkTransition helper function that would encapsulate the temporary updateAffectedByStarlarkTransition variable. Do you think that would resolve this comment also?

In addition to that, I could also extract one more helper method for the toHash calculation. E.g. something that could be invoked as:

```
TreeMap<String, Object> toHash =
       getAffectedOptionsWithCurrentValues(buildConfigOptions,
                                           optionInfoMap,
                                           toOptions);
```

What do you prefer?

Reply:
I don't think this method is long enough where we need to split it up too much, and I think the control flow for `toHash` is clear enough. So I'd be happy with just the `updateAffectedByStarlarkTransition` helper you suggested.

Reply:
OK! Resolved in 104f1dd by only extracting updateAffectedByStarlarkTransition.